cdap-common/src/main/java/io/cdap/cdap/common/internal/remote/DefaultRemoteAuthenticatorProvider.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.internal.remote;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.security.spi.authenticator.RemoteAuthenticator;
+
+import javax.annotation.Nullable;
+import javax.inject.Named;
+
+/**
+ * Provides a remote authenticator via the extension loader.
+ */
+public class DefaultRemoteAuthenticatorProvider implements Provider<RemoteAuthenticator> {
+
+  public static final String OVERRIDE_AUTHENTICATOR_CONFIG = "OverrideRemoteAuthenticatorConfig";
+
+  private final String defaultAuthenticatorName;
+  private final RemoteAuthenticatorExtensionLoader remoteAuthenticatorExtensionLoader;
+  private GceRemoteAuthenticator gceRemoteAuthenticator;
+
+  @Inject
+  DefaultRemoteAuthenticatorProvider(CConfiguration cConf,
+                                     @Named(OVERRIDE_AUTHENTICATOR_CONFIG) @Nullable String overrideAuthenticatorConfig,
+                                     RemoteAuthenticatorExtensionLoader remoteAuthenticatorExtensionLoader) {
+    // Use the override authenticator for subcomponents if it's set, but default to overall system wide configuration
+    // if no override is set.
+    if (overrideAuthenticatorConfig != null && cConf.get(overrideAuthenticatorConfig) != null) {
```

Comment:
Why not just bind the config key name to use? Essentially combining this `if-else` logic to the `RuntimeMonitors`.

Reply:
Can you clarify what you mean? In the current implementation, the override key name is bound, but the default config is used in situations in which the override is unbound or null. If the default was set as a binding, it would break in situations in which the override key was bound but not set.

Reply:
Discussed offline, updated the name to use `authenticatorNameKey` instead of `overrideAuthenticatorConfig` and added binding of the system-wide default key value to avoid binding a null value. Thanks!